app/src/WebsiteGenerator/TemplateRenderer/TwigExtension/SlidesEmbedExtension.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Phpsw\Website\WebsiteGenerator\TemplateRenderer\TwigExtension;
+
+use Phpsw\Website\Entity\Talk;
+use Twig_Extension;
+use Twig_SimpleFunction;
+
+class SlidesEmbedExtension extends Twig_Extension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return [
+            new Twig_SimpleFunction('embedSlides', [$this, 'getSlidesEmbed'], ['is_safe' => ['html']]),
+            new Twig_SimpleFunction('slidesLink', [$this, 'getSlidesLink'], ['is_safe' => ['html']]),
+        ];
+    }
+
+    /**
+     * Returns the string to embed the slides.
+     *
+     * How we embed them will differ depending on the source
+     *
+     * @param Talk $talk
+     *
+     * @return string
+     */
+    public function getSlidesEmbed(Talk $talk): string
+    {
+        $iFrameUrl = '';
+        $embedString = '';
+
+        if ('.pdf' == substr($talk->getSlidesUrl(), -4)) {
+            // PDFs can be embedded via an iFrame
+            $iFrameUrl = "https://mozilla.github.io/pdf.js/web/viewer.html?file={$talk->getSlidesUrl()}#zoom=page-fit";
+        } elseif (strpos($talk->getSlidesUrl(), 'slideshare')) {
+            // We use embedly to embed slideshare
+            $embedString = '<blockquote class="embedly-card"><h4><a href="'.$talk->getSlidesUrl().'">'.$talk->getTitle().'</a></h4></blockquote><script async src="//cdn.embedly.com/widgets/platform.js" charset="UTF-8"></script>';
```

Comment:
Do we want markup like this within PHP code? Or should it be split into different Twig partials or something else? Just seems a little weird.